### ISSUE TITLE
Upgrade 1.21.1 -> 1.22.0 breaks OIDC

### ISSUE THREAD

**Issue by klauserber** [workadventure/workadventure#4327]
Hello,

I have an issue when I upgrade my system from 1.21.1 to 1.22.0.

WA, Synapse, Jitsi and Dex are running in Kubernetes. In 1.21.1 the authentication via Dex with GitGub or Google is working fine. After the update I get this screen after the authentication in the identity provider (e. g. GitHub):

<img width="1574" alt="Screenshot 2024-10-15 at 22 37 57" src="https://github.com/user-attachments/assets/285dc4f7-1468-4078-937f-969701b54b7b">

I think it has somthing to do with this PR: https://github.com/workadventure/workadventure/pull/4295

Seems as if the javascript in the response gets only displayed an not executed. I cannot see any authToken in the local storage.

There seems to be no content/type header in the response. I do not have the error when a I disable Matrix.

**Comment by klauserber** [comment]
I see this errors when the play service starts in the log:

```
'Accept: application/json, text/plain, */*\r\n' +
'Authorization: Bearer syt_cm9vdA_qHAravILHESNrnbpmhhY_0dWHay\r\n' +
'cache-control: no-cache\r\n' +
'pragma: no-cache\r\n' +
'expires: 0\r\n' +
'User-Agent: axios/1.6.6\r\n' +
'Accept-Encoding: gzip, compress, deflate, br\r\n' +
'Host: synapse.synapse.svc.cluster.local:8008\r\n' +
'Connection: keep-alive\r\n' +
'\r\n',
_keepAliveTimeout: 0,
_onPendingData: [Function: nop],
agent: [Agent],
socketPath: undefined,
method: 'GET',
maxHeaderSize: undefined,
insecureHTTPParser: undefined,
joinDuplicateHeaders: undefined,
path: '/_matrix/client/r0/directory/room/%23current-visited-room:isium.de',
_ended: true,
res: [IncomingMessage],
aborted: false,
timeoutCb: null,
upgradeOrConnect: false,
parser: null,
maxHeadersCount: null,
reusedSocket: true,
host: 'synapse.synapse.svc.cluster.local',
protocol: 'http:',
_redirectable: [Writable],
[Symbol(kCapture)]: false,
[Symbol(kBytesWritten)]: 0,
[Symbol(kNeedDrain)]: false,
[Symbol(corked)]: 0,
[Symbol(kOutHeaders)]: [Object: null prototype],
[Symbol(errored)]: null,
[Symbol(kHighWaterMark)]: 16384,
[Symbol(kRejectNonStandardBodyWrites)]: false,
[Symbol(kUniqueHeaders)]: null
},
data: {
errcode: 'M_NOT_FOUND',
error: 'Room alias #current-visited-room:isium.de not found'
}
}
}
All static assets have been loaded!
SocketManager => handleJoinSpace => pingMessage
Error: ChatID is undefined
at SocketManager.leaveChatRoomArea (/usr/src/play/src/pusher/services/SocketManager.ts:1463:35)
at close (/usr/src/play/src/pusher/controllers/IoSocketController.ts:1059:35)
```

I have seen in the code that we need now a extra variable 'MATRIX_DOMAIN' and MATRIX_API_URI needs now a trailing '/'. But I still have these error on start of the play service.

And I still have the callback issue. There seems to be no content/type header in the response. I do not have the error when a I disable Matrix.

**Comment by moufmouf** [comment]
Gosh! Indeed, the screen is only used when Matrix integration is enabled. It allows us to save temporarily the authToken and get it back when Matrix authentication is done.

Previously, we used cookies for that but some SSO providers (Microsoft) are generating huge authTokens that don't always fit in a cookie.

You are right, the issue is probably linked to the fact we don't send the content/type header.

We'll fix this.

Out of curiosity, what browser are you using? (because it's working pretty well on all the browsers we have tested so far)
